be/src/formats/orc/orc_chunk_reader.cpp

```diff
@@ -1330,9 +1335,9 @@ static void _try_implicit_cast(TypeDescriptor* from, const TypeDescriptor& to) {
         // violated during creating a DecimalV3Column via ColumnHelper::create(...).
         if (t2 == PrimitiveType::TYPE_DECIMALV2) {
             from->type = t2;
-        } else if (from->precision > decimal_precision_limit<int64_t>) {
+        } else if ((from->precision > decimal_precision_limit<int64_t>) | (to.type == PrimitiveType::TYPE_DECIMAL128)) {
             from->type = PrimitiveType::TYPE_DECIMAL128;
-        } else if (from->precision > decimal_precision_limit<int32_t>) {
+        } else if ((from->precision > decimal_precision_limit<int32_t>) | (to.type == PrimitiveType::TYPE_DECIMAL64)) {
```

Comment:
```suggestion
        } else if ((from->precision > decimal_precision_limit<int32_t>) || (to.type == PrimitiveType::TYPE_DECIMAL64)) {
```
Use `||` when it is not in the vectorized scenario.